File: app/src/main/java/fr/frogdevelopment/nihongo/dico/ui/main/CustomTypefaceSpan.java
package fr.frogdevelopment.nihongo.dico.ui.main;

import android.graphics.Typeface;
import android.text.TextPaint;
import android.text.style.TypefaceSpan;

public class CustomTypefaceSpan extends TypefaceSpan {

    private final Typeface newType;

    public CustomTypefaceSpan(Typeface type) {
        super("family");
        newType = type;
    }

    @Override
    public void updateDrawState(TextPaint ds) {
        ds.setTypeface(newType);
    }

    @Override
    public void updateMeasureState(TextPaint paint) {
        paint.setTypeface(newType);
    }

}

File: app/src/main/java/com/frogdevelopment/nihongo/dico/ui/main/CustomTypefaceSpan.kt
package com.frogdevelopment.nihongo.dico.ui.main

import android.graphics.Typeface
import android.text.TextPaint
import android.text.style.TypefaceSpan

class CustomTypefaceSpan(private val newType: Typeface) : TypefaceSpan("japanese") {

    override fun updateDrawState(ds: TextPaint) {
        ds.typeface = newType
    }

    override fun updateMeasureState(paint: TextPaint) {
        paint.typeface = newType
    }

}